pkg/controllermanager/controller/project/project_activity_control.go

```diff
@@ -0,0 +1,101 @@
+// limitations under the License.
+
+package project
+
+import (
+	"context"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/logger"
+	gutil "github.com/gardener/gardener/pkg/utils/gardener"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NewProjectStaleReconciler creates a new instance of a reconciler which reconciles the LastActivityTimestamp of a project, then it calls the stale reconciler.
+func NewActivityReconciler(
+	l logrus.FieldLogger,
+	config *config.ProjectControllerConfiguration,
+	gardenClient client.Client,
+	projectStaleReconciler reconcile.Reconciler,
+) reconcile.Reconciler {
+	projectsLastActivityMap := make(map[string]metav1.Time)
+	return &projectActivityReconciler{
+		logger:                 l,
+		config:                 config,
+		gardenClient:           gardenClient,
+		projectsLastActivity:   projectsLastActivityMap,
+		projectStaleReconciler: projectStaleReconciler,
+	}
+}
+
+type projectActivityReconciler struct {
+	logger                 logrus.FieldLogger
+	gardenClient           client.Client
+	config                 *config.ProjectControllerConfiguration
+	projectsLastActivity   map[string]metav1.Time
+	projectStaleReconciler reconcile.Reconciler
+}
+
+func (r *projectActivityReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	shoot := &gardencorev1beta1.Shoot{}
+
+	if err := r.gardenClient.Get(ctx, client.ObjectKey{Namespace: request.Namespace, Name: request.Name}, shoot); err != nil {
+		logger.Logger.Errorf("Couldn't get Shoot due to: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	return r.reconcile(ctx, shoot)
+}
+
+func (r *projectActivityReconciler) reconcile(ctx context.Context, obj client.Object) (reconcile.Result, error) {
+	project, err := gutil.ProjectForNamespaceFromReader(ctx, r.gardenClient, obj.GetNamespace())
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{Requeue: false}, nil
+		}
+		logger.Logger.Errorf("Couldn't get project due to: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	if obj.GetCreationTimestamp().UTC().After(project.Status.LastActivityTimestamp.UTC()) {
+		if err := updateStatus(ctx, r.gardenClient, project, func() {
+			now := obj.GetCreationTimestamp()
+			project.Status.LastActivityTimestamp = &now
+		}); err != nil {
+			logger.Logger.Errorf("Error while updating project: %v", err)
+			return reconcile.Result{}, err
+		}
+		return r.projectStaleReconciler.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: project.Name, Namespace: project.Name}})
```

Comment:
Suggestion: Change the [projectUpdate](https://github.com/gardener/gardener/blob/83deabdcd595c7672a872ce8f811bf58c95d1d79/pkg/controllermanager/controller/project/project_control.go#L44) event handler to enqueue the project `c.projectStaleQueue.Add(key)` when `project.Status.LastActivityTimestamp` is updated. This way, you get rid of the `projectStaleReconciler` dependency.